### ISSUE TITLE
gmic-3.0.2: FTBFS due to double inclusion of CImg.h

### ISSUE THREAD

**Issue by chris2553** [GreycLab/gmic#366]
gmic-3.0.2 fails to build because CImg.h is including directly in src/gmic_cli.cpp as well is via gmic.h. This triggers the error at line 196 of gmic.h.
The following patch fixes the problem:

--- gmic-3.0.2/src/gmic_cli.cpp.orig    2022-01-20 09:16:29.704469598 +0000
+++ gmic-3.0.2/src/gmic_cli.cpp 2022-01-20 09:16:41.716470103 +0000
@@ -49,7 +49,6 @@
 */
 
 #include <signal.h>
-#include "CImg.h"
 #include "gmic.h"
 using namespace cimg_library;

**Comment by dtschump** [comment]
That is actually because the `CMakeList` file wrongly defined a macro `gmic_core` when compiling some files.
The source code is actually correct.
See PR https://github.com/dtschump/gmic/pull/364 that fix this issue.

PS : The file `src/Makefile` provided by default does not have this problem.

**Comment by yurivict** [comment]
Similar failure on FreeBSD:
```
In file included from /wrkdirs/usr/ports/graphics/gmic/work/gmic-3.0.2/src/gmic_cli.cpp:53:
/wrkdirs/usr/ports/graphics/gmic/work/gmic-3.0.2/src/gmic.h:196:2: error: "[gmic] *** Error *** File 'CImg.h' has been already included (should have been done first in file 'gmic.h')."
#error "[gmic] *** Error *** File 'CImg.h' has been already included (should have been done first in file 'gmic.h')."
 ^
```

**Comment by dtschump** [comment]
Should be fixed in latest release 3.2.0.
3.2.1 release is planned for tomorrow.
